packages/vm/src/buildBlock.ts

```diff
@@ -216,11 +229,7 @@ export class BlockBuilder {
    * Reverts the checkpoint on the StateManager to reset the state from any transactions that have been run.
    */
   async revert() {
-    this.checkStatus()
-    if (this.checkpointed) {
-      await this.vm.stateManager.revert()
-      this.reverted = true
-    }
+    this.blockStatus = { status: BuildStatus.Reverted }
```

Comment:
This removes a checkpoint mechanism check, what are the implications of this?

Reply:
This particular one shouldn't matter because once reverted, the builder can never be used again and will get deleted on the call to ForkChoiceUpdated. Since the underlying checkpoints are in a copy of state manager attached to the block builder, it gets thrown away so extra work to revert the checkpoints

Reply:
Does that matter? Once a block builder has reverted, should anyone care about the state manager attached to it? You can't do anything more with the builder at that point.

Reply:
I am also sceptical on this one or at least might not have the complete picture: if someone uses the same state for (other) VM tasks and block building this non-reverting of the StateManager state would definitely change the behavior - as Jochem pointed out - right? 🤔

What is/was the reason to remove this StateManager revert in the first place? Was this necessary respectively does this fix anything?

Reply:
Update: so I do not find this VM/StateManager copy in the VM itself (unless I am overlooking something), so from the perspective of the VM this is a breaking API (respectively: behavioral change) we should revert. Even in the case copy happens in the VM there still might be some use cases of the block builder (Hardhat using this for their tests or something and relying on this assumption of the state being reset) where this breaks current usages.

Let me know if I am talking non-sense here, but as I read between the lines in the chat this also hasn't been such a high profile change, rather some optimization? So in doubt we should be able to easily revert? 🤔

Reply:
the copy happens in client in `miner/miner.ts` for pre merge and for postmerge `miner/pendingBlock.ts`'s `start` recieves a `vm` copy,
but anyway added the revert logic back as suuggested :+1:

Reply:
Yes, but relevant for deciding what is breaking in the VM scope is the code inside of the VM package since people are using this stand-alone.